backend/src/zelthy/apps/auditlog/models.py

```diff
@@ -0,0 +1,605 @@
+import ast
+import contextlib
+import json
+from copy import deepcopy
+from datetime import timezone
+from typing import Any, Callable, Dict, List, Union
+
+from dateutil import parser
+from dateutil.tz import gettz
+from django.conf import settings
+from django.contrib.contenttypes.fields import GenericRelation
+from django.contrib.contenttypes.models import ContentType
+from django.core import serializers
+from django.core.exceptions import (
+    FieldDoesNotExist,
+    ObjectDoesNotExist,
+    ValidationError,
+    FieldError,
+)
+from django import apps
+from django.db import DEFAULT_DB_ALIAS, models
+from django.db.models import Q, QuerySet
+from django.utils import formats
+from django.utils import timezone as django_timezone
+from django.utils.encoding import smart_str
+from django.utils.translation import gettext_lazy as _
+
+from zelthy.apps.auditlog.diff import mask_str
+from zelthy.core.utils import get_current_request
+from zelthy.apps.object_store.models import ObjectStore
+from zelthy.apps.exportjob.models import ExportJob
+
+DEFAULT_OBJECT_REPR = "<error forming object repr>"
+
+
+class LogEntryManager(models.Manager):
+    """
+    Custom manager for the :py:class:`LogEntry` model.
+    """
+
+    def log_create(self, instance, force_log=False, **kwargs):
+        changes = kwargs.get("changes", None)
+        pk = self._get_pk_value(instance)
+
+        if changes is not None:
+            kwargs.setdefault(
+                "content_type", ContentType.objects.get_for_model(instance)
+            )
+            kwargs.setdefault("object_pk", pk)
+            kwargs.setdefault("object_repr", smart_str(instance))
+
+            if isinstance(pk, int):
+                kwargs.setdefault("object_id", pk)
+
+            get_additional_data = getattr(instance, "get_additional_data", None)
+            if callable(get_additional_data):
+                kwargs.setdefault("additional_data", get_additional_data())
+            db = instance._state.db
+            try:
+                obj = ObjectStore.objects.get(object_uuid=instance.object_uuid)
```

Comment:
also handle the case where objectstore doesn't contain the object with given `instance.object_uuid`, in this case code shouldn't break, object_ref can be None in that case